Tchap/Managers/User/UserService.swift

```diff
@@ -0,0 +1,97 @@
+/*
+ Copyright 2018 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+
+/// `UserService` implementation of `UserServiceType` is used to handle Tchap users.
+final class UserService: UserServiceType {
+    
+    // MARK: - Constants
+    
+    private enum Constants {
+        static let searchUsersLimit: UInt = 50
+    }
+    
+    // MARK: - Properties
+    
+    private let session: MXSession
+    
+    // MARK: - Setup
+    
+    init(session: MXSession) {
+        self.session = session
+    }
+    
+    // MARK: - Public
+    
+    func buildUser(from userId: String) -> User {
```

Comment:
I would copy this method as a private one.

I would change this public method to retrieve the MXUser from the session if any,
else return the result of the private method